File: ru.kmorozov.api.onedrive/src/main/java/ru/kmorozov/onedrive/client/authoriser/AuthorisationProvider.java
package ru.kmorozov.onedrive.client.authoriser;

import java.io.IOException;
import java.nio.file.Path;

public interface AuthorisationProvider {

    String getAccessToken() throws IOException;

    void refresh() throws IOException;

    class FACTORY {
        public static AuthorisationProvider create(final Path keyFile, String clientId, String clientSecret) throws IOException {
            return new OneDriveAuthorisationProvider(keyFile, clientId, clientSecret);
        }

        public static void printAuthInstructions(String clientId) {
            OneDriveAuthorisationProvider.printAuthInstructions(clientId);
        }
    }
}

File: ru.kmorozov.api.onedrive/src/main/java/ru/kmorozov/onedrive/client/authoriser/AuthorisationProvider.kt
package ru.kmorozov.onedrive.client.authoriser

import java.io.IOException
import java.nio.file.Path

interface AuthorisationProvider {

    val accessToken: String

    @Throws(IOException::class)
    fun refresh()

    object FACTORY {
        @Throws(IOException::class)
        fun create(keyFile: Path, clientId: String, clientSecret: String): AuthorisationProvider {
            return OneDriveAuthorisationProvider(keyFile, clientId, clientSecret)
        }

        fun printAuthInstructions(clientId: String) {
            OneDriveAuthorisationProvider.printAuthInstructions(clientId)
        }
    }
}
